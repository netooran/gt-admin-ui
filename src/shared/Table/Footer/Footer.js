import styled from 'styled-components';
import PropTypes from 'prop-types';
import Pagination from 'shared/Table/Pagination/Pagination';
import Button from 'shared/Button/Button';

const Wrapper = styled.div`
  display: flex;
  padding: 1rem;
  justify-content: space-between;
`;

const Footer = ({
  options,
  page,
  noOfpages,
  setPage,
  hasSelection,
  removeSelected,
}) => {
  const { deletable, paginatable } = options;
  const isTableNotEmpty = !!noOfpages;

  return (
    <Wrapper>
      {deletable && isTableNotEmpty && (
        <Button
          aria-label="Delete selected"
          variant="solid"
          importance="danger"
          disabled={!hasSelection}
          onClick={removeSelected}
        >
          Delete Selected
        </Button>
      )}
      {paginatable && isTableNotEmpty && (
        <Pagination page={page} noOfpages={noOfpages} goto={setPage} />
      )}
    </Wrapper>
  );
};

Footer.propTypes = {
  options: PropTypes.shape({
    deletable: PropTypes.bool,
    paginatable: PropTypes.bool,
  }),
  page: PropTypes.number,
  noOfpages: PropTypes.number,
  hasSelection: PropTypes.bool,
  setPage: PropTypes.func,
  removeSelected: PropTypes.func,
};

export default Footer;
