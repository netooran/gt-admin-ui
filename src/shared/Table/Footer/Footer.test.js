import { fireEvent, render, screen } from '@testing-library/react';
import Footer from './Footer';

test('renders footer with pagination', () => {
  const props = {
    options: { paginatable: true },
    page: 1,
    noOfpages: 2,
    setPage: jest.fn(),
  };
  const { container } = render(<Footer {...props} />);
  expect(container).toMatchSnapshot();

  fireEvent.click(screen.getByText('2'));
  expect(props.setPage).toHaveBeenCalledWith(2);
});

describe('Footer with select all actions', () => {
  test('renders delete selected button disabled', () => {
    const props = {
      options: { deletable: true },
      noOfpages: 1,
      hasSelection: false,
      removeSelected: jest.fn(),
    };

    const { container } = render(<Footer {...props} />);
    expect(container).toMatchSnapshot();

    fireEvent.click(screen.getByText('Delete Selected'));
    expect(props.removeSelected).not.toHaveBeenCalled();
  });

  test('renders delete selected button active', () => {
    const props = {
      options: { deletable: true },
      noOfpages: 1,
      hasSelection: true,
      removeSelected: jest.fn(),
    };

    const { container } = render(<Footer {...props} />);
    expect(container).toMatchSnapshot();

    fireEvent.click(screen.getByText('Delete Selected'));
    expect(props.removeSelected).toHaveBeenCalled();
  });

  test('renders no pagination & delete button given no page', () => {
    const props = {
      options: { deletable: true },
      noOfpages: 0,
      hasSelection: true,
      removeSelected: jest.fn(),
    };
    const { container } = render(<Footer {...props} />);
    expect(container).toMatchSnapshot();
  });
});
