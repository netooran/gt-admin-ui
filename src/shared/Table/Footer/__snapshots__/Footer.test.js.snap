// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer with select all actions renders delete selected button active 1`] = `
.c1 {
  border: none;
  cursor: pointer;
  opacity: 0.7;
  font-weight: bold;
  padding: 0.5rem 1.5rem;
  border-radius: 3rem;
  --primary-color: var(--danger-color);
  background-color: var(--primary-color);
  color: white;
}

.c1 svg {
  width: 1.2rem;
}

.c1:disabled {
  opacity: 0.4;
  pointer-events: none;
}

.c1:hover {
  opacity: 1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div>
  <div
    class="c0"
  >
    <button
      aria-label="Delete selected"
      class="c1"
    >
      Delete Selected
    </button>
  </div>
</div>
`;

exports[`Footer with select all actions renders delete selected button disabled 1`] = `
.c1 {
  border: none;
  cursor: pointer;
  opacity: 0.7;
  font-weight: bold;
  padding: 0.5rem 1.5rem;
  border-radius: 3rem;
  --primary-color: var(--danger-color);
  background-color: var(--primary-color);
  color: white;
}

.c1 svg {
  width: 1.2rem;
}

.c1:disabled {
  opacity: 0.4;
  pointer-events: none;
}

.c1:hover {
  opacity: 1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div>
  <div
    class="c0"
  >
    <button
      aria-label="Delete selected"
      class="c1"
      disabled=""
    >
      Delete Selected
    </button>
  </div>
</div>
`;

exports[`Footer with select all actions renders no pagination & delete button given no page 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`renders footer with pagination 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  border: none;
  cursor: pointer;
  opacity: 0.7;
  font-weight: bold;
  padding: 0.5rem 1.5rem;
  border-radius: 3rem;
  background-color: var(--primary-color);
  color: white;
  min-width: 3rem;
  min-height: 3rem;
  padding: 0;
}

.c2 svg {
  width: 1.2rem;
}

.c2:disabled {
  opacity: 0.4;
  pointer-events: none;
}

.c2:hover {
  opacity: 1;
}

.c5 {
  border: none;
  cursor: pointer;
  opacity: 0.7;
  font-weight: bold;
  padding: 0.5rem 1.5rem;
  border-radius: 3rem;
  background-color: transparent;
  border: 1px solid var(--primary-color);
  color: var(--primary-color);
  min-width: 3rem;
  min-height: 3rem;
  padding: 0;
}

.c5 svg {
  width: 1.2rem;
}

.c5:disabled {
  opacity: 0.4;
  pointer-events: none;
}

.c5:hover {
  opacity: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 button {
  margin-right: 0.5rem;
}

.c3 {
  --button-primary-color: #1890ff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        aria-label="Go to first page"
        class="c2 c3"
        disabled=""
        shape="round"
      >
        <svg
          aria-hidden="true"
          class="c4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.121 12l4.94-4.939-2.122-2.122L3.879 12l7.06 7.061 2.122-2.122z"
          />
          <path
            d="M17.939 4.939L10.879 12l7.06 7.061 2.122-2.122L15.121 12l4.94-4.939z"
          />
        </svg>
      </button>
      <button
        aria-label="Go to previous page"
        class="c2 c3"
        disabled=""
        shape="round"
      >
        <svg
          aria-hidden="true"
          class="c4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.939 4.939L6.879 12l7.06 7.061 2.122-2.122L11.121 12l4.94-4.939z"
          />
        </svg>
      </button>
      <button
        aria-label="Go to page 1"
        class="c5 c3"
        shape="round"
      >
        1
      </button>
      <button
        aria-label="Go to page 2"
        class="c2 c3"
        shape="round"
      >
        2
      </button>
      <button
        aria-label="Go to next page"
        class="c2 c3"
        shape="round"
      >
        <svg
          aria-hidden="true"
          class="c4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.061 19.061L17.121 12l-7.06-7.061-2.122 2.122L12.879 12l-4.94 4.939z"
          />
        </svg>
      </button>
      <button
        aria-label="Go to last page"
        class="c2 c3"
        shape="round"
      >
        <svg
          aria-hidden="true"
          class="c4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.061 4.939l-2.122 2.122L15.879 12l-4.94 4.939 2.122 2.122L20.121 12z"
          />
          <path
            d="M6.061 19.061L13.121 12l-7.06-7.061-2.122 2.122L8.879 12l-4.94 4.939z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
